var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -87,79 +88,83 @@ class Dealii(CMakePackage, CudaPackage):
                when='@8.5.0:~mpi+python')
-    depends_on("boost@1.59.0:1.63,1.66:+thread+system+serialization+iostreams+mpi+python",
+    depends_on('boost@1.59.0:1.63,1.65.0,1.66:+thread+system+serialization+iostreams+mpi+python',
                when='@8.5.0:+mpi+python')
     # bzip2 is not needed since 9.0
-    depends_on("bzip2", when='@:8.99')
-    depends_on("lapack")
-    depends_on("muparser")
-    depends_on("suite-sparse")
-    depends_on("tbb")
-    depends_on("zlib")
+    depends_on('bzip2', when='@:8.99')
+    depends_on('lapack')
+    depends_on('muparser')
+    depends_on('suite-sparse')
+    depends_on('tbb')
+    depends_on('zlib')
 
     # optional dependencies
-    depends_on("mpi",              when="+mpi")
-    depends_on("adol-c@2.6.4:",    when='@9.0:+adol-c')
-    depends_on("arpack-ng+mpi",    when='+arpack+mpi')
-    depends_on("assimp",           when='@9.0:+assimp')
-    depends_on("doxygen+graphviz", when='+doc')
-    depends_on("graphviz",         when='+doc')
-    depends_on("gsl",              when='@8.5.0:+gsl')
-    depends_on("hdf5+mpi+hl",      when='+hdf5+mpi')
-    depends_on("cuda@8:",          when='+cuda')
-    depends_on("cmake@3.9:",       when='+cuda')
+    depends_on('mpi',              when='+mpi')
+    depends_on('adol-c@2.6.4:',    when='@9.0:+adol-c')
+    depends_on('arpack-ng+mpi',    when='+arpack+mpi')
+    depends_on('assimp',           when='@9.0:+assimp')
+    depends_on('doxygen+graphviz', when='+doc')
+    depends_on('graphviz',         when='+doc')
+    depends_on('gsl',              when='@8.5.0:+gsl')
+    depends_on('hdf5+mpi+hl',      when='+hdf5+mpi')
+    depends_on('cuda@8:',          when='+cuda')
+    depends_on('cmake@3.9:',       when='+cuda')
     # currently deal.II does not build with Cmake 3.10, see
     # https://github.com/dealii/dealii/issues/5510
-    depends_on("cmake@:3.9.99")
+    depends_on('cmake@:3.9.99')
     # FIXME: concretizer bug. The two lines mimic what comes from PETSc
     # but we should not need it
-    depends_on("metis@5:+int64+real64",   when='+metis+int64')
-    depends_on("metis@5:~int64+real64",   when='+metis~int64')
-    depends_on("nanoflann",        when="@9.0:+nanoflann")
-    depends_on("netcdf+mpi",       when="+netcdf+mpi")
-    depends_on("netcdf-cxx",       when='+netcdf+mpi')
-    depends_on("oce",              when='+oce')
-    depends_on("p4est",            when='+p4est+mpi')
-    depends_on("petsc+mpi~int64",  when='+petsc+mpi~int64')
-    depends_on("petsc+mpi+int64",  when='+petsc+mpi+int64')
-    depends_on("petsc@:3.6.4",     when='@:8.4.1+petsc+mpi')
+    depends_on('metis@5:+int64+real64',   when='+metis+int64')
+    depends_on('metis@5:~int64+real64',   when='+metis~int64')
+    depends_on('nanoflann',        when='@9.0:+nanoflann')
+    depends_on('netcdf+mpi',       when='+netcdf+mpi')
+    depends_on('netcdf-cxx',       when='+netcdf+mpi')
```

Comment:
I see from your modifications to the `netcdf-cxx` package thats its actually deprecated. Do we not support the newer bindings provided by `netcdf-cxx4` ?

Reply:
I was curious as well, let me try to build...

Reply:
Thanks for investigating :-)

Reply:
according to https://github.com/dealii/dealii/blob/master/cmake/modules/FindNETCDF.cmake#L40 we are looking for old c++ lib (without `4` at the end), will try to build anyway...

Reply:
nope, we don't support it. We have 
```
#include <netcdfcpp.h>
```
in `grid_in.cc` and this file is not there in `netcdf-cxx4`.

Reply:
Ah, ok. Well, thanks for looking into this anyway!